### DISCUSSION TITLE
"Show persona card on hover" setting not working on some accounts

### DISCUSSION THREAD

**Discussion by mariaarcena** [microsoft-search/pnp-modern-search#1772] [Q&A]
I feel like I've been asking many dumb questions but this time I feel like this is a bug...? I'm not sure yet so I will make this a discussion post first.

So I was doing some testing and I noticed that on some accounts I have this problem where when the I turn **off** the "Show persona card on hover", the user pop up
![image](https://user-images.githubusercontent.com/17426994/151641079-a9a8d6f1-ffb8-4148-a6b2-8cdac7626539.png)

But I turn **on** the "Show persona card on hover", the user is returned but shows only an icon and the card doesn't pop out
![image](https://user-images.githubusercontent.com/17426994/151641101-681e8786-3e9b-4b91-820f-5eaa752f97aa.png)

This bug only appears on accounts that have emails that are different from their usernames/user principal names. **Is there some explanation as to why this is happening?**

My wandering thoughts:
- Can this possibly because of the data being used when it attempts to use the api that pulls the persona card (since the persona card uses microsoft graph, correct?)
- Related to the bullet above: can it be because the email and username for this account are different, it leads to having problems calling the persona card?
- Do we know what data element/attribute is being used to called the persona card that pops up?

**Comment by wobba** [comment]
@mariaarcena it's because the Persona card uses UPN for the default setup in the `user-id` field. You can change the template to use `persona-query` instead which can then query on the e-mail address when that is the data you provide.

Looking at the latest version the template uses  `<mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">` where the `PersonQuery` slot is mapped to `UserName`. So it should in theory be set up correct. `UserEmail` is also mapped to the `UserName` managed property by default.

For native accounts it works as expected for me, but I get the avatar only for external accounts - which is expected.

**Reply by mariaarcena** [reply]
@wobba Okay! I will play with that and check it out! Quick question though, may I understand the set up of this: 
`<mgt-person user-id="{{getUserEmail (slot item  #@root.slots.PersonQuery)}}" person-card="hover">`

Since you are trying to access PersonQuery from the chosen slots (the setting where we get to add more data slots), I understand what is inside the parentheses. But why does it have `getUserEmail` before it? 

![image](https://user-images.githubusercontent.com/17426994/151875811-f4406346-7dfa-44b7-a181-97e4948407ae.png)

**Reply by wobba** [reply]
To strip out the upn/address from the property used which can have other "noise" around it. Which is typically what you see in xOWSUser managed properties.

**Reply by mariaarcena** [reply]
Update!

So initially I thought I had to go to Search Results settings > Available layouts > Select "People" > then edit the "Edit results template". So I did trial and error of editing "PersonQuery" to WorkEmail or Email or UserEmail 😅🤭 I even trying editing the getUserEmail to other things too!

![image](https://user-images.githubusercontent.com/17426994/152443658-898d72c6-5b27-4f66-a445-a0be6131655e.png)

But I only realized what you meant of "You can change the template to use persona-query instead which can then query on the e-mail address when that is the data you provide."

So I went from Search Results settings > Layout slots > "Customize" under Edit layout slots for this data source and change this (below) "UserName"
![image](https://user-images.githubusercontent.com/17426994/152444189-19100354-d965-4a9f-ac38-8973f6e3109d.png)

into this "WorkEmail"
![image](https://user-images.githubusercontent.com/17426994/152444293-41e2d9c4-4d14-4d43-8b95-993d1a3b7f8c.png)

but of course in order to do the above, I had to add "WorkEmail" to the Import/Export settings:
![image](https://user-images.githubusercontent.com/17426994/152444520-0827fbb7-1615-4af6-922e-138ebf6a27cd.png)

The person now appears nicely and their card pop out as well
![image](https://user-images.githubusercontent.com/17426994/152444884-9e36a743-a9de-4aa3-bcf9-61e66367b51a.png)

![image](https://user-images.githubusercontent.com/17426994/152445199-99d7587e-092f-4ed5-a6eb-e5748b0051f4.png)

So again, this is a solution if your organizations modifies accounts where the usernames end up different from their emails. So like for example, if a person initially started with a username and email `pretty.person@whatever.org` and got their username changed to `beautiful.person@whatever.org` (through whatever reason) and their email stayed as `pretty.person@whatever.org`; this is where the solution above will fix that!

Thank you @wobba for all your help! 😊

**Reply by wobba** [reply]
So you changed from `user-id` to `person-query` and then used `WorkEmail`, which makes sense :) Still weird that the `user-id=AccountName` fails as that is more specific compared to the e-mail address as an e-mail address can be associated with multiple accounts.

**Reply by Impakt-101** [reply]
Having the same/similar issue. Turning on Show persona card on hover the user is returned but shows only an icon and the card doesn't pop out.

When I try to add "WorkEmail" to the Import/Export settings it doesn't show as an option when going to Search Results settings > Layout slots > "Customize" under Edit layout slots for this data source.

 Also, PersonQuery is set to "AADObjectID" by default instead of "UserName"
